sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/ExpressionConverter.java

```diff
@@ -486,27 +489,38 @@ private RexNode convertRexNodeFromComputedColumnWithFieldList(
         return rexBuilder()
             .makeInputRef(fieldList.get(windowFieldIndex).getType(), windowFieldIndex);
       case FIXED_WINDOW_END:
-        // WINDOW END is a function call
         operands.add(
             rexBuilder().makeInputRef(fieldList.get(windowFieldIndex).getType(), windowFieldIndex));
         // TODO: check window_end 's duration is the same as it's aggregate window.
         operands.add(
             convertIntervalToRexIntervalLiteral(
                 (ResolvedLiteral) functionCall.getArgumentList().get(0)));
-        return rexBuilder().makeCall(SqlStdOperatorTable.PLUS, operands);
+        return rexBuilder()
+            .makeCall(createSqlFunction("timestamp_add", SqlTypeName.TIMESTAMP), operands);
       case SLIDING_WINDOW_END:
         operands.add(
             rexBuilder().makeInputRef(fieldList.get(windowFieldIndex).getType(), windowFieldIndex));
         operands.add(
             convertIntervalToRexIntervalLiteral(
                 (ResolvedLiteral) functionCall.getArgumentList().get(1)));
-        return rexBuilder().makeCall(SqlStdOperatorTable.PLUS, operands);
+        return rexBuilder()
+            .makeCall(createSqlFunction("timestamp_add", SqlTypeName.TIMESTAMP), operands);
       default:
         throw new RuntimeException(
             "Does not support window start/end: " + functionCall.getFunction().getName());
     }
   }
 
+  private SqlFunction createSqlFunction(String name, SqlTypeName returnType) {
```

Comment:
Can you add this to `SqlOperators.java` and add the two calls above as a constant there instead?

Reply:
Done!